core/corehttp/proxy_test.go

```diff
@@ -0,0 +1,34 @@
+package corehttp
+
+import (
+	"net/http"
+	"strings"
+	"testing"
+
+	"github.com/ipfs/go-ipfs/thirdparty/assert"
+)
+
+func TestParseRequest(t *testing.T) {
+	url := "http://localhost:5001/p2p/QmT8JtU54XSmC38xSb1XHFSMm775VuTeajg7LWWWTAwzxT/http/path/to/index.txt"
+	req, _ := http.NewRequest("GET", url, strings.NewReader(""))
+
+	parsed, err := parseRequest(req)
+	if err != nil {
+		t.Error(err)
```

Comment:
this should be `t.Fatal`
```suggestion
		t.Fatal(err)
```